src/bin/testnode.rs

```diff
@@ -129,8 +141,24 @@ fn main() {
         serve_sock.local_addr().unwrap(),
     );
     eprintln!("starting server...");
-    let threads = rpu.serve(d, serve_sock, gossip_sock, exit.clone(), stdout())
-        .unwrap();
+    let threads = if matches.opt_present("r") {
```

Comment:
Let's try not to merge this today. Once I split the RPU into RPU/TPU, then this code will make no sense, and it should be easier to rebase than for me to unwind it and refactor. In the new version, but leader and validator will have the same RPU, and you'll be able to pass this option to the TPU to configure it. Should be no need for a TVU at all, since the new TPU will be a linear pipeline. Just swap out (or configure) a stage or two for validation. Keep an eye on this one: https://github.com/solana-labs/solana/pull/220